server/api/game/controller.js

```diff
@@ -135,12 +137,36 @@ class Game {
     this.gameState.players[playerId].direction = direction;
   };
 
+  hasGameEnded = () => {
+    const { players } = this.gameState;
+    const playerStatus = Object.keys(players).map(player => player.alive);
+    const { isPlayerOneAlive, isPlayerTwoAlive } = playerStatus;
+    return this.hasFoodFinished() || (!isPlayerOneAlive && !isPlayerTwoAlive);
+  };
+
+  hasFoodFinished = () => {
+    const { gridState } = this.gameState;
+    let food = false;
+    gridState.forEach((row) => {
+      row.forEach((cell) => {
+        if (cell === entityToCode('food')) {
+          food = true;
+        }
+      });
+    });
+    return food;
+  };
+
   startGame = () => {
     this.setInitialGameState();
     // eslint-disable-next-line no-console
     console.log('Game started');
     this.gameState.interval = setInterval(() => {
       this.calculateNextGameState();
+      const gameEndStatus = this.hasGameEnded();
+      if (gameEndStatus) {
+        // io.in(this.roomId).emit(GAME_OVER);
```

Comment:
Why is this line commented?

Reply:
Because I didn't know that `io` is global.